### ISSUE TITLE
named vectors supplied as palettes get translated to keyed js objects instead of arrays

### ISSUE THREAD

**Issue by MilesMcBain** [anthonynorth/rdeck#17]
Map breaks with:

```
react.js:9 TypeError: e.palette.map is not a function
    at L (rdeck.js:1)
    at rdeck.js:1
    at Array.map (<anonymous>)
    at rdeck.js:1
    at new x (rdeck.js:1)
    at create (rdeck.js:1)
    at Array.map (<anonymous>)
    at V (rdeck.js:1)
    at ro (react.js:9)
    at Hu (react.js:9)
```

R does warn:

```
Input to asJSON(keep_vec_names=TRUE) is a named vector. In a future version of jsonlite, this option will not be supported, and named vectors will be translated into arrays instead of objects. If you want JSON object output, please use a named list instead. See ?toJSON.
```

**Comment by MilesMcBain** [comment]
could be my version of jsonlite?

- Session info --------------------------------------------------------------------------------------------------------------------------------------------------
 setting  value
 version  R version 4.0.2 (2020-06-22)
 os       Windows 10 x64
 system   x86_64, mingw32
 ui       RTerm
 language (EN)
 collate  English_Australia.1252
 ctype    English_Australia.1252
 tz       Australia/Brisbane
 date     2020-12-15

- Packages ------------------------------------------------------------------------------------------------------------------------------------------------------   
 package     * version    date       lib source
 assertthat    0.2.1      2019-03-21 [1] CRAN (R 4.0.2)
 backports     1.1.10     2020-09-15 [1] CRAN (R 4.0.3)
 blob          1.2.1      2020-01-20 [1] CRAN (R 4.0.2)
 broom         0.7.0      2020-07-09 [1] CRAN (R 4.0.2)
 callr         3.5.1      2020-10-13 [1] CRAN (R 4.0.3)
 capsule     * 0.1.1.9000 2020-11-02 [1] local
 cellranger    1.1.0      2016-07-27 [1] CRAN (R 4.0.2)
 class         7.3-17     2020-04-26 [2] CRAN (R 4.0.2)
 classInt      0.4-3      2020-04-07 [1] CRAN (R 4.0.2)
 cli           2.2.0      2020-11-20 [1] CRAN (R 4.0.2)
 codetools     0.2-16     2018-12-24 [2] CRAN (R 4.0.2)
 colorspace    2.0-0      2020-11-11 [1] CRAN (R 4.0.3)
 conflicted  * 1.0.4      2019-06-21 [1] CRAN (R 4.0.2)
 crayon        1.3.4      2017-09-16 [1] CRAN (R 4.0.2)
 curl          4.3        2019-12-02 [1] CRAN (R 4.0.2)
 data.table    1.13.0     2020-07-24 [1] CRAN (R 4.0.2)
 DBI           1.1.0      2019-12-15 [1] CRAN (R 4.0.2)
 dbplyr        1.4.4      2020-05-27 [1] CRAN (R 4.0.2)
 desc          1.2.0      2018-05-01 [1] CRAN (R 4.0.2)
 devtools      2.3.1      2020-07-21 [1] CRAN (R 4.0.2)
 digest        0.6.27     2020-10-24 [1] CRAN (R 4.0.3)
 dotenv      * 1.0.2      2017-03-01 [1] CRAN (R 4.0.0)
 dplyr       * 1.0.2      2020-08-18 [1] CRAN (R 4.0.2)
 e1071         1.7-4      2020-10-14 [1] CRAN (R 4.0.3)
 ellipsis      0.3.1      2020-05-15 [1] CRAN (R 4.0.2)
 evaluate      0.14       2019-05-28 [1] CRAN (R 4.0.2)
 fansi         0.4.1      2020-01-08 [1] CRAN (R 4.0.2)
 forcats     * 0.5.0      2020-03-01 [1] CRAN (R 4.0.2)
 fs            1.5.0      2020-07-31 [1] CRAN (R 4.0.2)
 generics      0.1.0      2020-10-31 [1] CRAN (R 4.0.2)
 ggplot2     * 3.3.2      2020-06-19 [1] CRAN (R 4.0.2)
 glue        * 1.4.2      2020-08-27 [1] CRAN (R 4.0.2)
 gtable        0.3.0      2019-03-25 [1] CRAN (R 4.0.2)
 haven         2.3.1      2020-06-01 [1] CRAN (R 4.0.2)
 hms           0.5.3      2020-01-08 [1] CRAN (R 4.0.2)
 htmltools     0.5.0      2020-06-16 [1] CRAN (R 4.0.2)
 htmlwidgets * 1.5.3      2020-12-10 [1] CRAN (R 4.0.3)
 httr          1.4.2      2020-07-20 [1] CRAN (R 4.0.2)
 igraph        1.2.5      2020-03-19 [1] CRAN (R 4.0.2)
 jsonlite      1.7.2      2020-12-09 [1] CRAN (R 4.0.3)
 KernSmooth    2.23-17    2020-04-26 [2] CRAN (R 4.0.2)
 knitr       * 1.30       2020-09-22 [1] CRAN (R 4.0.3)
 lifecycle     0.2.0      2020-03-06 [1] CRAN (R 4.0.2)
 lubridate   * 1.7.9.2    2020-11-13 [1] CRAN (R 4.0.3)
 magrittr      2.0.1      2020-11-17 [1] CRAN (R 4.0.3)
 memoise       1.1.0      2017-04-21 [1] CRAN (R 4.0.2)
 modelr        0.1.8      2020-05-19 [1] CRAN (R 4.0.2)
 munsell       0.5.0      2018-06-12 [1] CRAN (R 4.0.2)
 pillar        1.4.7      2020-11-20 [1] CRAN (R 4.0.2)
 pkgbuild      1.1.0      2020-07-13 [1] CRAN (R 4.0.2)
 pkgconfig     2.0.3      2019-09-22 [1] CRAN (R 4.0.2)
 pkgload       1.1.0      2020-05-29 [1] CRAN (R 4.0.2)
 prettyunits   1.1.1      2020-01-24 [1] CRAN (R 4.0.2)
 processx      3.4.4      2020-09-03 [1] CRAN (R 4.0.3)
 ps            1.4.0      2020-10-07 [1] CRAN (R 4.0.3)
 purrr       * 0.3.4      2020-04-17 [1] CRAN (R 4.0.2)
 qfes        * 0.5.0.9000 2020-11-01 [1] Github (qfes/qfes@aa52317)
 R6            2.5.0      2020-10-28 [1] CRAN (R 4.0.2)
 ragg        * 0.3.1      2020-07-03 [1] CRAN (R 4.0.2)
 Rcpp          1.0.5      2020-07-06 [1] CRAN (R 4.0.2)
 rdeck       * 0.2.6.9000 2020-12-15 [1] Github (anthonynorth/rdeck@e9c7a09)
 readr       * 1.4.0      2020-10-05 [1] CRAN (R 4.0.3)
 readxl        1.3.1      2019-03-13 [1] CRAN (R 4.0.2)
 remotes       2.2.0      2020-07-21 [1] CRAN (R 4.0.2)
 reprex        0.3.0      2019-05-16 [1] CRAN (R 4.0.2)
 rlang         0.4.9      2020-11-26 [1] CRAN (R 4.0.3)
 rmarkdown   * 2.3        2020-06-18 [1] CRAN (R 4.0.2)
 rprojroot     2.0.2      2020-11-15 [1] CRAN (R 4.0.3)
 rvest         0.3.6      2020-07-25 [1] CRAN (R 4.0.2)
 scales        1.1.1      2020-05-11 [1] CRAN (R 4.0.2)
 sessioninfo   1.1.1      2018-11-05 [1] CRAN (R 4.0.2)
 sf          * 0.9-6      2020-09-13 [1] CRAN (R 4.0.2)
 snakecase     0.11.0     2019-05-25 [1] CRAN (R 4.0.2)
 stringi       1.5.3      2020-09-09 [1] CRAN (R 4.0.2)
 stringr     * 1.4.0      2019-02-10 [1] CRAN (R 4.0.2)
 stylebox    * 0.1.0      2020-08-17 [1] Github (anthonynorth/stylebox@500218e)
 systemfonts   0.2.3      2020-06-09 [1] CRAN (R 4.0.2)
 tarchetypes * 0.0.0.9001 2020-11-24 [1] Github (wlandau/tarchetypes@d8e1760)
 targets     * 0.0.0.9003 2020-11-24 [1] Github (wlandau/targets@89be7e5)
 testthat      3.0.0      2020-10-31 [1] CRAN (R 4.0.3)
 tibble      * 3.0.4      2020-10-12 [1] CRAN (R 4.0.3)
 tidyr       * 1.1.2      2020-08-27 [1] CRAN (R 4.0.2)
 tidyselect    1.1.0      2020-05-11 [1] CRAN (R 4.0.2)
 tidyverse   * 1.3.0      2019-11-21 [1] CRAN (R 4.0.2)
 units         0.6-7      2020-06-13 [1] CRAN (R 4.0.2)
 usethis       1.6.3      2020-09-17 [1] CRAN (R 4.0.3)
 utf8          1.1.4      2018-05-24 [1] CRAN (R 4.0.2)
 uuid          0.1-4      2020-02-26 [1] CRAN (R 4.0.0)
 vctrs         0.3.5      2020-11-17 [1] CRAN (R 4.0.3)
 withr         2.3.0      2020-09-22 [1] CRAN (R 4.0.3)
 xfun          0.19       2020-10-30 [1] CRAN (R 4.0.3)
 xml2          1.3.2      2020-04-23 [1] CRAN (R 4.0.2)
 yaml          2.2.1      2020-02-01 [1] CRAN (R 4.0.2)

[1] C:/Users/msmcbain/libs/R
[2] C:/R/R-4.0.2/library

**Comment by anthonynorth** [comment]
Will be fixed in 0.3

**Comment by anthonynorth** [comment]
closed in #18
